From c582cf0c732af342305d3710f793d66f464fa58d Mon Sep 17 00:00:00 2001
From: Nolan Stover <nstover@fusionbox.com>
Date: Fri, 10 Jul 2015 15:48:05 -0600
Subject: [PATCH] Use `setsid` function from Python's `os` module

---
 src/watchdog/tricks/__init__.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/watchdog/tricks/__init__.py b/src/watchdog/tricks/__init__.py
index 3bda5fc..5c377c7 100644
--- a/src/watchdog/tricks/__init__.py
+++ b/src/watchdog/tricks/__init__.py
@@ -138,13 +138,14 @@ def __init__(self, command, patterns=None, ignore_patterns=None,
                  kill_after=10):
         super(AutoRestartTrick, self).__init__(
             patterns, ignore_patterns, ignore_directories)
-        self.command = ['setsid'] + command
+        self.command = command
         self.stop_signal = stop_signal
         self.kill_after = kill_after
         self.process = None
 
     def start(self):
-        self.process = subprocess.Popen(self.command)
+
+        self.process = subprocess.Popen(self.command, preexec_fn=os.setsid)
 
     def stop(self):
         if self.process is None:
